readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -837,6 +839,229 @@ def test_command_oom_kill(self):
+            '--cache-dir',
+            mock.ANY,  # cache path
+        ]
+        args.extend(requirements)
+        self.build_env_mock.run.assert_called_once_with(
+            *args, bin_path=mock.ANY
+        )
+
+    def test_install_user_requirements(self):
+        self.build_env_mock.project = self.project_sphinx
+        self.build_env_mock.version = self.version_sphinx
+        python_env = Virtualenv(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock
+        )
+
+        checkout_path = python_env.checkout_path
+        docs_requirements = os.path.join(
+            checkout_path, 'docs', 'requirements.txt'
+        )
+        root_requirements = os.path.join(
+            checkout_path, 'requirements.txt'
+        )
+        paths = {
+            os.path.join(checkout_path, 'docs'): True,
+        }
+        args = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '--exists-action=w',
+            '--cache-dir',
+            mock.ANY,  # cache path
+            'requirements_file'
+        ]
+
+        paths[docs_requirements] = True
+        paths[root_requirements] = False
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(docs_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        paths[docs_requirements] = False
+        paths[root_requirements] = True
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(root_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        paths[docs_requirements] = True
+        paths[root_requirements] = True
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        args[-1] = '-r{}'.format(docs_requirements)
+        self.build_env_mock.run.assert_called_with(
+            *args, cwd=mock.ANY, bin_path=mock.ANY
+        )
+
+        paths[docs_requirements] = False
+        paths[root_requirements] = False
+        with fake_paths_lookup(paths):
+            python_env.install_user_requirements()
+        self.build_env_mock.run.assert_not_called()
+
+    def test_install_core_requirements_sphinx_conda(self):
+        python_env = Conda(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock,
+        )
+        python_env.install_core_requirements()
+        conda_requirements = [
+            'mock',
+            'pillow',
+            'sphinx',
+            'sphinx_rtd_theme',
+        ]
+        pip_requirements = [
+            'recommonmark',
+            'readthedocs-sphinx-ext',
+        ]
+
+        args_pip = [
+            'python',
+            mock.ANY,  # pip path
+            'install',
+            '-U',
```

Comment:
You can also use a shared `args_pip` for all these tests. You will need to change the `-U` by `--upgrade` but I think that's OK and it's better to use long attributes instead of short. So, :+1:

Reply:
There is two types of the arg_pip, on conda the `--use-wheel` isn't used. It's that on purpose? or perhaps we can unify both :)?